db/migrate/20170421071137_migrate_query_custom_field_filters.rb

```diff
@@ -0,0 +1,79 @@
+##
+# Updates filter values for list custom fields so that they refer
+# to the newly introduced CustomOption IDs. If no custom option can
+# be found for a filter value the value remains so it can be tried
+# to restore or map it by hand.
+class MigrateQueryCustomFieldFilters < ActiveRecord::Migration[5.0]
+  def up
+    update_query_filters! do |filter|
```

Comment:
I'll probably regret it later but I suggest not validating here as well. When I tried to run the migration on my anonymized production data, the migration didn't get very far before stumbling over the first invalid query. (e.g. #5399)

As it is not the objective of this migration to fix all of those errors, it is probably ok to forfeit the validations.